### PR TITLE
Add purchases for CUBE study labs

### PR DESCRIPTION
**Pull request by mrgnr**
## Done

The "Prepare" button now displays the purchase modal if study labs have not yet been purchased. Otherwise, it links to the corresponding study lab.

Depends on https://github.com/canonical-web-and-design/ubuntu.com/pull/10689

## QA

- Run the server with `dotrun`
- Go to http://localhost:8001/cube/microcerts?test_backend=true
- Click on the "Prepare" button and complete the purchase process
- Once the page reloads, click on the "Prepare" button to go to the study lab
- Go to https://qa.cube.ubuntu.com/dashboard and unenroll yourself from the "CUBE 2020 Study Labs" course (click on the gear icon and select "Unenroll")
- Go to http://localhost:8001/cube/study?test_backend=true
- Click on the "Purchase study labs access" button and complete the purchase process
- Once the page reloads, click on the "Access study labs" button to go to the study labs

## Issue / Card

Fixes https://github.com/canonical-web-and-design/commercial-squad/issues/348

### /cube/microcerts

cube-prepare

### /cube/study
cube-study

### CODE REVIEW

**Comment by wgx** [comment]
@mrgnr 

>Given the CUBE site's current structure, the button on https://ubuntu.com/cube/microcerts SHOULD say "Prepare", as it leads to the free study material. The button on https://ubuntu.com/cube/study should say "Purchase", because that's where customers can decide to buy the labs. There is copy on that page explaining what they are for.

This change (4) from Lex sounds ideal, please do implement this too.

**Comment by mrgnr** [comment]
@wgx

> 1. The "M" icon is filled - yet the one I have passed has a different icon, is this a bug:

I'll look into this.

> 2. Upon a purchase, the page reloads but the newly purchased item still says "purchase" - known issue?

It seems that demos.haus uses the production edX instance rather than QA, so I think this PR needs to be tested locally.

> 3. We need some copy for this section:

Updated (I'd accidentally removed what what there before :laughing: )

> This change (4) from Lex sounds ideal, please do implement this too.

Updated.

**Comment by carkod** [comment]
> @wgx
> 
> > 1. The "M" icon is filled - yet the one I have passed has a different icon, is this a bug:
> 
> I'll look into this.
> 
> > 1. Upon a purchase, the page reloads but the newly purchased item still says "purchase" - known issue?
> 
> It seems that demos.haus uses the production edX instance rather than QA, so I think this PR needs to be tested locally.
> 
> > 1. We need some copy for this section:
> 
> Updated (I'd accidentally removed what what there before laughing )
> 
> > This change (4) from Lex sounds ideal, please do implement this too.
> 
> Updated.

Point 2 looks like a backend issue, but I can't reproduce it. I just gave it a try with this PR and it does actually change the buttons, maybe it's been fixed already?

**Comment by carkod** [comment]
> @wgx
> 
> > 1. The "M" icon is filled - yet the one I have passed has a different icon, is this a bug:
> 
> I'll look into this.
> 
> > 2. Upon a purchase, the page reloads but the newly purchased item still says "purchase" - known issue?
> 
> It seems that demos.haus uses the production edX instance rather than QA, so I think this PR needs to be tested locally.
> 
> > 3. We need some copy for this section:
> 
> Updated (I'd accidentally removed what what there before laughing )
> 
> > 4. This change (4) from Lex sounds ideal, please do implement this too.
> 
> Updated.

Issues 1 and 2 potentially will be fixed in https://github.com/canonical-web-and-design/ubuntu.com/pull/10788

**Review by ClementChaumel — APPROVED** [review]

File: static/js/src/cube/hooks/useMicrocertsData.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from "react";
+import { Module } from "../types";
+
+const useMicrocertsData = () => {
+  const [modules, setModules] = useState<Module[]>([]);
+  const [studyLabs, setStudyLabs] = useState<Record<string, unknown>>({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState("");
+
```
[the remark this answers is not in the document]
Comment (mrgnr) [reply static/js/src/cube/hooks/useMicrocertsData.tsx]: Thanks for the suggestion! Done.

**Pull request merged**